lib/moment/processMomentsInBatches.js

```diff
@@ -38,18 +41,40 @@ export async function processMomentsInBatches(network, momentEvents) {
       );
 
       if (validMoments.length > 0) {
-        // Ensure all defaultAdmins exist in the artists table
-        const adminAddresses = validMoments.map((moment) => moment.defaultAdmin);
-        await ensureArtists(adminAddresses);
+        // Check which moments already exist in the database
+        const momentAddresses = validMoments.map((moment) => moment.address);
+        const existingTokens = await selectTokens(momentAddresses, "address");
```

Comment:
You're currently looking up if the moment already exists in supabase, right?
- that's still going to be a lot of `egress` (GET returns all that data)
- why not just make a variable for `blockNumber` and update your graphQL query to only fetch `blockNumber > cachedBlockNumber`?